pkg/segment/structs/evaluationstructs.go

```diff
@@ -1710,7 +1710,25 @@ func (self *TextExpr) EvaluateText(fieldToValue map[string]utils.CValueEnclosure
 	case "getfields":
 		fallthrough
 	case "typeof":
-		return "", fmt.Errorf("TextExpr.EvaluateText: dose not support functions:%v: right now", self.Op)
+		valueExpr, err := self.Val.EvaluateToString(fieldToValue)
+		if err != nil {
+			// Could not evaluate as string, try as float
+			_, floatErr := self.Val.EvaluateToFloat(fieldToValue)
+			if floatErr == nil {
+				return "Number", nil
+			}
+			if isBoolean(valueExpr) {
+				return "Boolean", nil
+			}
+			if isNull(valueExpr) {
+				return "Null", nil
+			}
+			return "Invalid", nil
+		}
+		if isNumber(valueExpr) {
+			return "Number", nil
+		}
+		return "String", nil
```

Comment:
1. I could be wrong, but I think if it fails to evaluate to a string, it will never succeed evaluating to a float
2. Don't you want the `isBoolean` and `IsNull` checks to be outside the `if err != nil` block?

Reply:
1. Yes, I think you're right but this comment on [L672](https://github.com/siglens/siglens/blob/3a413b9716e2a8365e8effb0f1d27643369fab9d/pkg/segment/structs/evaluationstructs.go#L672) says otherwise. However, I updated the logic to get the datatype from `utils.CValueEnclosure` instead of parsing it to find the datatype in case of field names like `eval res=typeof(city)`. If case the command has a raw value like `eval res=typeof(23) ` or `eval res =typeof("city")` then `CValueEnclosure` would have datatype as invalid and then we perform the parsing on the raw value to find the datatype.  Also, I have added testcases to `ingest.csv` for the strings and numbers, for null and boolean sigclient doesn't ingest these datatypes so I tested it manually. 
<img width="1450" alt="image" src="https://github.com/siglens/siglens/assets/113487319/3fc45ff6-523a-4c7f-a4a7-5ddb3867d5b6">